File: clients/java1.8_client/client/src/main/java/core/BaseStrategy.java
package core;

import core.API.Debug;
import core.API.Elevator;
import core.API.Passenger;

import java.util.List;

/**
 * Created by Boris on 01.02.17.
 */
public class BaseStrategy {
    private Debug debug;

    public void setDebug(Debug debug) {
        this.debug = debug;
    }
    public void log(Object object) {
        debug.log(object);
    }

    public void onTick(List<Passenger> myPassengers, List<Elevator> myElevators, List<Passenger> enemyPassengers, List<Elevator> enemyElevators) {
    }
}

File: clients/kotlin_client/client/src/main/kotlin/core/BaseStrategy.kt
package core

import core.API.Debug
import core.API.Elevator
import core.API.Passenger

open class BaseStrategy {
  private var debug: Debug? = null

  fun setDebug(debug: Debug) {
    this.debug = debug
  }
  fun log(obj: Any) = debug?.log(obj)
  open fun onTick(myPassengers: List<Passenger>, myElevators: List<Elevator>, enemyPassengers: List<Passenger>, enemyElevators: List<Elevator>) {}
}